qiime2/sdk/plugin_manager.py

```diff
@@ -101,43 +122,112 @@ def _integrate_plugin(self, plugin):
             self.formats[name] = record
         self.type_formats.extend(plugin.type_formats)
 
+    def get_semantic_types(self):
+        types = set()
+
+        for plugin in self.plugins.values():
+            for type_record in plugin.types:
+                types.add(type_record)
+
+        return types
+
     # TODO: Should plugin loading be transactional? i.e. if there's
     # something wrong, the entire plugin fails to load any piece, like a
     # databases rollback/commit
 
-    @property
-    def importable_formats(self):
-        """Return formats that are importable.
+    def get_formats(self, *, filter=None, semantic_type=None):
+        """
+        get_formats(self, *, filter=None, semantic_type=None)
+
+        filter : enum
+            filter is an enum integer that will be used to determine user
+            input to output specified formats
 
-        A format is importable in a QIIME 2 deployment if it can be transformed
-        into at least one of the canonical semantic type formats.
+        semantic_type : TypeExpression | String
+            The semantic type is used to filter the formats associated with
+            that specific semantic type
 
+        This method will filter out the formats using the filter provided by
+        the user and the semantic type. The return is a set of filtered
+        formats.
         """
-        importable_formats = {}
-        for name, record in self.formats.items():
-            from_type = transform.ModelType.from_view_type(
-                record.format)
+        if filter is not None and filter not in GetFormatFilters:
+            raise ValueError("The format filter provided: %s is not "
+                             "valid.", (filter))
+
+        if semantic_type is None:
+            formats = set(f.format for f in self.type_formats)
+
+        else:
+            formats = set()
+
+            if isinstance(semantic_type, str):
+                semantic_type = parse_type(semantic_type, "semantic")
```

Comment:
you may want to verify that non-strings are in fact semantic types, otherwise we'll get a weird error with `not_type <= type_format.type_expression` downstream.